core/sub.mk

```diff
@@ -5,8 +5,22 @@ subdirs-y += drivers
 ifeq ($(CFG_WITH_USER_TA)-$(CFG_REE_FS_TA),y-y)
 gensrcs-y += ta_pub_key
 produce-ta_pub_key = ta_pub_key.c
-depends-ta_pub_key = $(TA_SIGN_KEY)
+depends-ta_pub_key = $(TA_SIGN_KEY) scripts/pem_to_pub_c.py
 recipe-ta_pub_key = scripts/pem_to_pub_c.py --prefix ta_pub_key \
 		--key $(TA_SIGN_KEY) --out $(sub-dir-out)/ta_pub_key.c
 cleanfiles += $(sub-dir-out)/ta_pub_key.c
 endif
+
+ifeq ($(CFG_WITH_USER_TA)-$(CFG_EARLY_TA),y-y)
+define process_early_ta
+early-ta-$1-uuid := $(firstword $(subst ., ,$(notdir $1)))
+gensrcs-y += early-ta-$1
+produce-early-ta-$1 = early_ta_$$(early-ta-$1-uuid).c
+depends-early-ta-$1 = $1 scripts/ta_bin_to_c.py
+recipe-early-ta-$1 = scripts/ta_bin_to_c.py --compress --ta $1 \
+		--out $(sub-dir-out)/early_ta_$$(early-ta-$1-uuid).c
+cleanfiles += $(sub-dir-out)/early_ta_$$(early-ta-$1-uuid).c
+cflags-remove-early-ta-$1-y := -pedantic
+endef
+$(foreach f, $(EARLY_TA), $(eval $(call process_early_ta,$(f))))
```

Comment:
s/`EARLY_TA`/`EARLY_TA_PATHS`/

Could you check that this works ok even if `EARLY_TA_PATHS` is empty or even not defined ?